sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDbInstance.scala

```diff
@@ -0,0 +1,434 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming.state
+
+import java.io.File
+import java.util.Locale
+
+import org.apache.commons.io.FileUtils
+import org.rocksdb._
+import org.rocksdb.RocksDB
+import org.rocksdb.util.SizeUnit
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.Utils
+
+class RocksDbInstance(keySchema: StructType, valueSchema: StructType, identifier: String)
+    extends Logging {
+
+  import RocksDbInstance._
+  RocksDB.loadLibrary()
+
+  protected var db: RocksDB = null
+  protected var dbPath: String = _
+  protected val readOptions: ReadOptions = new ReadOptions()
+  protected val writeOptions: WriteOptions = new WriteOptions()
+  protected val table_options = new BlockBasedTableConfig
+  protected val options: Options = new Options()
+
+  def isOpen(): Boolean = {
+    db != null
+  }
+
+  def open(path: String, conf: Map[String, String], readOnly: Boolean): Unit = {
+    verify(db == null, "Another rocksDb instance is already actve")
+    try {
+      setOptions(conf)
+      db = readOnly match {
```

Comment:
Wouldn't it be easier to say
```
if (readOnly) {
...
} else {
...
}
```

Reply:
Your suggestion totally makes sense. I went overboard in using match..case construct. Will make the changes.